### PR TITLE
support query multiple reports at once

### PR DESCRIPTION
**Pull request by Roy-Hu**

### CODE REVIEW

File: report.go [hunk 1]
```diff
@@ -59,3 +59,77 @@ func GetReportOID(c *Client, link *Link, oid OID) ([]USAReport, error) {
 	}
 	return reports, err
 }
+
+func GetMultiReports(c *Client, link *Link, urrids []uint64, seids []uint64) ([]USAReport, error) {
```
Comment (tim-ywliu) [inline report.go]:
```
// map[uint64][]uint32 // key: seid, value: urrids
func GetMultiReports(c *Client, link *Link, map[uint64][]uint32) ([]USAReport, error) {
```

File: attr_report.go [hunk 2]
```diff
@@ -95,9 +102,14 @@ func DecodeVolumeMeasurement(b []byte) (VolumeMeasurement, error) {
 			v := native.Uint64(b[n:])
 			VolMeasurement.DownlinkPktNum = v
 			VolMeasurement.Flag |= DLNOP
+		case UR_VOLUME_MEASUREMENT_END:
```
Comment (tim-ywliu) [inline attr_report.go]:
```
default:
    break
```

File: attr_report.go [hunk 3]
```diff
@@ -63,6 +69,7 @@ type VolumeMeasurement struct {
 
 func DecodeVolumeMeasurement(b []byte) (VolumeMeasurement, error) {
```
Comment (tim-ywliu) [inline attr_report.go]:
```
func DecodeVolumeMeasurement(b []byte) (VolumeMeasurement, []byte, error) {
```

File: attr_report.go [hunk 4]
```diff
@@ -95,9 +102,14 @@ func DecodeVolumeMeasurement(b []byte) (VolumeMeasurement, error) {
...
+		case UR_VOLUME_MEASUREMENT_END:
+			VMEnd = true
 		}
 
 		b = b[hdr.Len.Align():]
+		if VMEnd {
+			break
+		}
 	}
 	return VolMeasurement, nil
```
Comment (tim-ywliu) [inline attr_report.go]:
```
return VolMeasurement, b, nil
```

File: attr_report.go [hunk 5]
```diff
@@ -153,9 +166,17 @@ func DecodeUSAReport(b []byte) (*USAReport, error) {
 		case UR_END_TIME:
 			v := native.Uint64(b[n:])
 			report.EndTime = time.Unix(0, int64(v))
+		case UR_SEID:
+			report.SEID = native.Uint64(b[n:])
```
Comment (tim-ywliu) [inline attr_report.go:158]:
```
		case UR_SEID:
			report.SEID = native.Uint64(b[n:])
			urAttrEnd = true
```

File: attr_report.go [hunk 6]
```diff
@@ -153,9 +166,17 @@ func DecodeUSAReport(b []byte) (*USAReport, error) {
...
+		// The UR_END prevents iterate to the next UR netlink attirbute
+		case UR_END:
+			UREnd = true
 		}
 
 		b = b[hdr.Len.Align():]
+		if UREnd {
```
Comment (tim-ywliu) [inline attr_report.go]:
```
if urAttrEnd
```

File: attr_report.go [hunk 7]
```diff
@@ -63,6 +69,7 @@ type VolumeMeasurement struct {
 
 func DecodeVolumeMeasurement(b []byte) (VolumeMeasurement, error) {
 	var VolMeasurement VolumeMeasurement
+	VMEnd := false
```
Comment (tim-ywliu) [inline attr_report.go]: remove it

File: attr_report.go [hunk 8]
```diff
@@ -127,6 +139,7 @@ func DecodeAllUSAReports(b []byte) ([]USAReport, error) {
 
 func DecodeUSAReport(b []byte) (*USAReport, error) {
 	report := new(USAReport)
+	UREnd := false
```
Comment (tim-ywliu) [inline attr_report.go]:
```
urAttrEnd := false
```

File: attr_report.go [hunk 9]
```diff
@@ -153,9 +166,17 @@ func DecodeUSAReport(b []byte) (*USAReport, error) {
 		case UR_END_TIME:
 			v := native.Uint64(b[n:])
 			report.EndTime = time.Unix(0, int64(v))
+		case UR_SEID:
+			report.SEID = native.Uint64(b[n:])
+		// The UR_END prevents iterate to the next UR netlink attirbute
+		case UR_END:
```
Comment (tim-ywliu) [inline attr_report.go]: remove it

File: attr_report.go (hunk 2 above)
Comment (Roy-Hu) [reply attr_report.go]: break seems meaningless here, so I move return to here

File: attr_report.go [hunk 10]
```diff
@@ -61,13 +63,12 @@ type VolumeMeasurement struct {
 	DownlinkPktNum uint64
 }
 
-func DecodeVolumeMeasurement(b []byte) (VolumeMeasurement, error) {
+func DecodeVolumeMeasurement(b []byte) (VolumeMeasurement, []byte, error) {
```
Comment (tim-ywliu) [inline attr_report.go]: why return []byte? It seems no use.

**Review by tim-ywliu — APPROVED** [review]

**Pull request merged**